plugins/inputs/wireless/wireless_mac_test.go

```diff
@@ -0,0 +1,78 @@
+//go:build darwin
+// +build darwin
+
+package wireless
+
+import (
+	"runtime"
+	"testing"
+)
+
+func TestLoadWirelessTable(t *testing.T) {
+	goOS := runtime.GOOS
+	ns := Wireless{}
+	ns.DumpZeros = true
+	if goOS == "darwin" {
+		// line of input
+		macInput := `agrCtlRSSI: -42
+     agrExtRSSI: 0
+    agrCtlNoise: -92
+    agrExtNoise: 0
+          state: running
+        op mode: station
+     lastTxRate: 300
+        maxRate: 450
+lastAssocStatus: 0
+    802.11 auth: open
+      link auth: wpa2-psk
+          BSSID:
+           SSID: Foo Bar
+            MCS: 15
+        channel: 157,1`
+		// the headers we expect from that line of input
+
+		// the map of data we expect.
+		macParsed := map[string]interface{}{
+			"agrCtlRSSI":      int64(-42),
+			"agrExtRSSI":      int64(0),
+			"agrCtlNoise":     int64(-92),
+			"agrExtNoise":     int64(0),
+			"lastTxRate":      int64(300),
+			"maxRate":         int64(450),
+			"lastAssocStatus": int64(0),
+			"MCS":             int64(15),
```

Comment:
Sample config output mentions `dump_zeros` is enabled by default, so I wouldn't expect the 0 values in here. You could add another test where the setting is disabled and have these 0 fields.

Reply:
I don't see this reflected yet. Did you push your changes?